File: 70.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        if n == 2:
            return 2
        dp = [0 for _ in range(n+1)]
        dp[1] = 1
        dp[2] = 2
        for x in range(3,n+1):
            dp[x] = dp[x-2] + dp[x-1]
        return dp[n]



if __name__ == '__main__':
	a = Solution()
	print(a.climbStairs(12))

File: 70.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let m = (n+1) as usize;
		if n < 3{
			return n;
		}
		let mut dp = (vec![0; m]).into_boxed_slice();
		dp[1] = 1;
		dp[2] = 2;
		for i in 3..m{
			dp[i] = dp[i-1] + dp[i-2];
		}
		return dp[n as usize];
    }
}